test: provide capability for tests to run in their own namespace

ianvernon opened the pull request:
Running each set of tests in a `Describe` in its own namespace instead of the `default` namespace ensures that resource leakage between tests w.r.t. policies, services, and other associated resources stored in K8s is kept to a minimum. It also allows for deletion of all resources in a namespace with one command at the end of a run of a given set of tests. To allow for this functionality, a variety of changes had to be made:
    
* add variadic arguments for providing namespace to existing commands for backwards compatibility.
* change namespace parameter of `CiliumPolicyAction` to apply the policy in the namespace parameter, and hardcode the namespace used to get Cilium pods to be `kube-system`. Change callers accordingly.
    
One given set of tests within the `K8sPolicyTest` have been changed to use a specific namespace. Misc. fixes / changes needed for said tests to pass were added as well. Other tests can be migrated over time to use their own namespaces.

Signed-off by: Ian Vernon <ian@cilium.io>

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/8600)
<!-- Reviewable:end -->

Commit:
test: provide capability for tests to run in their own namespace

Running each set of tests in a `Describe` in its own namespace instead of the
`default` namespace ensures that resource leakage between tests w.r.t. policies,
services, and other associated resources stored in K8s is kept to a minimum. It
also allows for deletion of all resources in a namespace with one command at
the end of a run of a given set of tests. To allow for this functionality,
a variety of changes had to be made:

* add variadic arguments for providing namespace to existing commands for
backwards compatability.
* change namespace parameter of `CiliumPolicyAction` to apply the policy in the
namespace parameter, and hardcode the namespace used to get Cilium pods to be
`kube-system`. Change callers accordingly.

One given set of tests within the `K8sPolicyTest` have been changed to use a
specific namespace. Misc. fixes / changes needed for said tests to pass were
added as well. Other tests can be migrated over time to use their own
namespaces.

Signed-off by: Ian Vernon <ian@cilium.io>

nebril commented on test/k8sT/Policies.go:
```diff
@@ -87,16 +87,17 @@ var _ = Describe("K8sPolicyTest", func() {
+			ciliumPod        string
+			clusterIP        string
+			appPods          map[string]string
+			namespaceForTest = "basictest"
```
Can we set this based on `ginkgo.CurrentGinkgoTestDescription().FullTestText`? You would probably need to replace spaces and make it lowercase, but it might help avoid test namespaces being duplicated by mistake.

ianvernon replied:
Added!

nebril reviewed (dismissed):
LGTM aside from namespace name comment.

Commit:
test: add namespace generator function

This allows for each test to not have to hardcode its own namespace.

Signed-off by: Ian Vernon <ian@cilium.io>
